scipy/stats/stats.py

```diff
@@ -3294,26 +3294,31 @@ def spearmanr(a, b=None, axis=0, nan_policy='propagate'):
     """
     a, axisout = _chk_asarray(a, axis)
 
-    contains_nan, nan_policy = _contains_nan(a, nan_policy)
+    a_contains_nan, nan_policy = _contains_nan(a, nan_policy)
 
-    if contains_nan and nan_policy == 'omit':
+    if a_contains_nan and nan_policy == 'omit':
         a = ma.masked_invalid(a)
-        b = ma.masked_invalid(b)
```

Comment:
I think matching R's behaviour is always a good idea, but all this business of "missing values" seems to me a bit mathematical flakey in the first place (I read R's manual and the logic of possible options and behaviours is appalling).